clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -134,10 +194,19 @@ public int replicaId() {
         return replicaId;
     }
 
+    @Deprecated
```

Comment:
Already suggested, but we can continue using `PartitionData` and just deprecate the `maxNumOffsets` field.